lib/locale_threads.t

```diff
@@ -54,67 +471,1305 @@ SKIP: { # perl #127708
+                                    };
+                }
+            }
+
+            return \@combined;
+        } # End of gen_combinations() definition
+
+        # The result of calling gen_combinations() will be an array of hashes.
+        #
+        # The main value in each hash is an array (whose key is 'op_results')
+        # containing all the tests for this category for a thread.  If there
+        # were N calls to 'add_trial' for this category, there will be 'N'
+        # elements in the array.  Each element is a string packed with the
+        # operation to eval in a thread and the operation's expected result.
+        #
+        # The other data structure in each hash is an array with the key
+        # 'locales'.  That array is a list of every locale which yields the
+        # identical results in 'op_results'.
+        #
+        # Effectively, each hash gives all the tests for this category for a
+        # thread.  The total array of hashes gives the complete list of
+        # distinct tests possible on this system.  So later, a thread will
+        # pluck the next available one from the array..
+        my $combinations_ref = gen_combinations(\@ops, \%results,
+                                                $distincts{$category});
+
+        # Fix up the entries ...
+        foreach my $test ($combinations_ref->@*) {
+
+            # Sort the locale names; this makes it work for later comparisons
+            # to look at just the first element of each list.
+            $test->{locales}->@* =
+                                sort sort_by_hashed_locale $test->{locales}->@*;
+
+            # And for each test, calculate and store how many locales have the
+            # same result (saves recomputation later in a sort).  This adds
+            # another data structure to each hash in the main array.
+            my @individual_tests = $test->{op_results}->@*;
+            my @in_common_locale_counts;
+            foreach my $this_test (@individual_tests) {
+
+                # Each test came from %distincts, and there we have stored the
+                # list of all locales that yield the same result
+                push @in_common_locale_counts,
+                        scalar $distincts{$category}{$this_test}{locales}->@*;
+            }
+            push $test->{in_common_locale_counts}->@*, @in_common_locale_counts;
+        }
+
+        # Make a copy
+        my @cat_tests = $combinations_ref->@*;
+
+        # This sorts the test cases so that the ones with the least overlap
+        # with other cases are first.
+        sub sort_test_order {
+            my $a_tests_count = scalar $a->{in_common_locale_counts}->@*;
+            my $b_tests_count = scalar $b->{in_common_locale_counts}->@*;
+            my $tests_count = ($a_tests_count <= $b_tests_count)
+                              ? $a_tests_count
+                              : $b_tests_count;
```

Comment:
You define min() above, this could use `min($a_tests_count, $b_tests_count)`

Reply:
Thanks, done